atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/SweepableCellsReadWriteTest.java

```diff
@@ -96,6 +97,20 @@ public void canReadMultipleEntriesInSingleShardSameTransactionOneDedicated() {
         assertThat(reader.getLatestWrites(TS_REF, fixedShardAndStrategy)).hasSameElementsAs(writes);
     }
 
+    @Test
+    public void canReadMultipleEntriesInSingleShardMultipleTransactionsCombined() {
+        List<WriteInfo> writesFirst = writeToUniqueCellsInSameShard(writer, TS, MAX_CELLS_GENERIC * 2 + 1, true);
+        List<WriteInfo> writesLast = writeToUniqueCellsInSameShard(writer, TS + 2, 1, true);
+        List<WriteInfo> writesMiddle = writeToUniqueCellsInSameShard(writer, TS + 1, MAX_CELLS_GENERIC + 1, true);
+        List<WriteInfo> expectedResult = new ArrayList<>(writesLast);
+        expectedResult.addAll(writesMiddle.subList(writesLast.size(), writesMiddle.size()));
+        expectedResult.addAll(writesFirst.subList(writesMiddle.size(), writesFirst.size()));
+
+        ShardAndStrategy fixedShardAndStrategy = conservative(writesFirst.get(0).toShard(SHARDS));
+        List<WriteInfo> result = reader.getLatestWrites(TS_REF, fixedShardAndStrategy);
+        assertThat(result).hasSameElementsAs(expectedResult);
+    }
```

Comment:
This test is clever, though I've got a couple of comments:
- maybe `writeToUniqueCellsInSameShard` should be renamed to `writeToUniqueCellsInShardZero`?
- a comment on what the concrete list we build up actually is would be nice!
- I tried rewriting this to avoid the sublists, but the result wasn't pretty:

```
Collection<WriteInfo> expectedResult2 =
                Stream.of(writesFirst, writesLast, writesMiddle)
                .flatMap(List::stream)
                .collect(Collectors.groupingBy(WriteInfo::writeRef, Collectors.collectingAndThen(
                        Collectors.maxBy(Comparator.comparingLong(WriteInfo::timestamp)), Optional::get)))
                        .values();
```